src/charm.py

```diff
@@ -509,7 +68,11 @@ def _on_get_outstanding_certificate_requests_action(self, event: ActionEvent) ->
         Returns:
             None
         """
-        event.set_results({"Result": self.tls_certificates.get_requirer_csrs_with_no_certs()})
+        if not self._relation_created("certificates"):
+            event.fail(message="No certificates relation has been created yet.")
+            return None
```

Comment:
Suggesting to create a small method for this method as it's repeated in 86-88 and 107-109.

Reply:
It would only save us evet.fail() line one time (the rest would be the same) in exchange of creating an extra function that takes event as a parameter.
What do you think, is it needed?